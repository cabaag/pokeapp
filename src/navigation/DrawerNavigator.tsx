import { createDrawerNavigator } from '@react-navigation/drawer';
import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import { Image } from 'react-native';
import { enableScreens } from 'react-native-screens';
import MainScreen from '../screens/MainScreen/MainScreen';
import SettingsScreen from '../screens/SettingsScreen/SettingsScreen';

enableScreens();
const Drawer = createDrawerNavigator();

// eslint-disable-next-line react/display-name
const LogoTitle = memo(() => (
  <Image
    source={require('../assets/images/pokemon.png')}
    style={{ width: 140, height: 50 }}
  />
))

export default function DrawerNavigator(): React.ReactElement {
  const { t } = useTranslation()
  return (
    <Drawer.Navigator
      initialRouteName="Pomemons"
      screenOptions={{
        headerShown: true,
        headerStyle: {
          backgroundColor: '#3F51B5',
        },
        headerTintColor: '#fff',
        // eslint-disable-next-line react/display-name
        headerTitle: () => <LogoTitle />,
        headerTitleAlign: 'center',
      }}
    >
      <Drawer.Screen component={MainScreen} name="Pokemons" />
      <Drawer.Screen component={MainScreen} name={t('berries')} />
      <Drawer.Screen component={SettingsScreen} name={t('settings')} />
    </Drawer.Navigator>

  );
}

