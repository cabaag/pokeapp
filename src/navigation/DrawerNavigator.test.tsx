import { shallow, ShallowWrapper,  } from 'enzyme';
import React from 'react';
import DrawerNavigator from './DrawerNavigator';

describe('<DrawerNavigator />' , () => {
  const wrapper: ShallowWrapper = shallow(<DrawerNavigator />);

  it ('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})