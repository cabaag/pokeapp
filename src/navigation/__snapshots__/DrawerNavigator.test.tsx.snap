// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerNavigator /> match snapshot 1`] = `
<DrawerNavigator
  initialRouteName="Pomemons"
  screenOptions={
    Object {
      "headerShown": true,
      "headerStyle": Object {
        "backgroundColor": "#3F51B5",
      },
      "headerTintColor": "#fff",
      "headerTitle": [Function],
      "headerTitleAlign": "center",
    }
  }
>
  <Screen
    component={[Function]}
    name="Pokemons"
  />
  <Screen
    component={[Function]}
    name="berries"
  />
  <Screen
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    name="settings"
  />
</DrawerNavigator>
`;
